docs/DeveloperGuide.md

```diff
@@ -6,32 +6,215 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Implemented Features
 
+Following a restucture to the project to follow a more Object Oriented structure, as well as to allow for conformity and easy editing of code, many of our commands will follow a similar structure of the following:
+
+1. User enters a command, which is given to a Command Handler
+2. The handler passes this to a corresponding Parser object for that command, which will split it into its necessary parts
+3. The Parser will create a corresponding Command object for that command, which will perform its necessary actions
+4. The Command object will return successfully to the Command Handler, and the output will be given to the user (output for the user is stored in a `Response` class)
+
+The command to add a project is one example of this structure:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProjectNew.png)
+
+**Step1.** After recieving the necessary method call, the `CommandHandler` class will self-call executeCommand with both the list of projects as well as a constructor call to AddProjectCommandParser (and a call to the parse method of AddProjectCommandParser) as its parameters
+
+**Step2.** The AddProjectCommandParser object instance is created
+
+**Step3.** The parse method of AddProjectCommandParser is called with `commands` (the user input) as its parameter.
+
+**Step4.** AddProjectCommandParser will self call its `getProjectTitle` method. This will split the user's input and return the correct name for the project
+
+**Step5.** After returning this projectTitle to itself, `AddProjectCommandParser` will call the constructor for a new `AddProjectCommand` object
+
+**Step6.** This `AddProjectCommand` object will use the newly gathered project name to call the `addProjct` command of the `ProjectList` class.
+
+**Step7.** The status of this command completing is returned to `CommandHandler`, and the output is sent to the user.
+
+Deleting a project will follow nearly the exact same structure; the only difference is that the corresponding Parser and Command objects will be used (`DeleteProjectCommandParser` and `DeleteProjectCommand`)
+
+####Delete a Project
+![image info](./UmlDiagrams/deleteProjectNew.png)
```

Comment:
Same comments as the sequence diagram above

![image](https://user-images.githubusercontent.com/69310031/160963337-77407d74-332c-43f3-86fb-91f9c4b599f9.png)

![image](https://user-images.githubusercontent.com/69310031/160963357-4cecef7f-5063-4a1a-95dd-c3509b370647.png)

Reply:
Shouldn't arrow head touch the tip of the activation bar. Furthermore, shouldn't the floating arrows be removed?